Add support for setting a connection deadline

carlfriedrich opened the pull request:
This is useful if you do not want to keep a tunnel open for an unlimited time and have Sish taking care of closing it automatically.

carlfriedrich commented:
FYI: I did not find a place in the docs where all the connection parameters are described, so I added the documentation to the cheatsheet section.

antoniomika commented:
Hey @carlfriedrich 

Thanks for the contribution! I'll review this more thoroughly in a bit.

At first glance, let's use a pointer to a time.Time value and use a check for it being nil instead of the zero check. Also maybe accept epoch?

Best,

carlfriedrich commented:
@antoniomika Thanks for your quick reply again! Good idea with the pointer and the epoch value, I added that. Also refactored the parsing code into a dedicated function, that makes the code more readable IMO.

Looking forward to your detailed review.

antoniomika reviewed (approved):
Lgtm 👍

antoniomika commented:
Thanks again @carlfriedrich! Keep the good ideas coming :)

carlfriedrich commented:
@antoniomika Great, thanks again for maintaining this great project so reliably! 😎
